boards/opentitan/src/tests/otbn.rs

```diff
@@ -81,10 +82,26 @@ extern "C" {
     static _eapps: u8;
 }
 
+/// Static init an OtbnTestCallback, with
+/// respective buffers allocated
+/// Note: static_init!() returns an &'static mut reference.
+unsafe fn static_init_test_cb() -> &'static OtbnTestCallback {
+    let output_buf = static_init!([u8; 1024], [0; 1024]);
+    let temp_buf = static_init!([u8; 4], [0; 4]);
```

Comment:
You have identified a bug in the current code here.

The `TEMP` buffer is only ever used with `load_data()`. Previously `load_data()` took a `'static mut` as it was an async (like most calls in Tock). It was recently converted to just be a synchronous call as it's only used for the OTBN and it loads the data synchronously.

This means that `TEMP` and the new `temp_buf` don't need to be static. So you can remove this and just create a local array in `otbn_run_rsa_binary()` with

```rust
let temp_buf = [0; 4];
```

and just use that directly